osu.Game/OsuGame.cs

```diff
@@ -432,11 +438,47 @@ public bool OnPressed(GlobalAction action)
                 case GlobalAction.ToggleDirect:
                     direct.ToggleVisibility();
                     return true;
+                case GlobalAction.TakeScreenshot:
+                    if (Window.ScreenshotTakenAction == null)
+                        Window.ScreenshotTakenAction = (screenshotBitmap) =>
+                        {
+                            var fileName = getScreenshotFileName(screenshotFormat);
+
+                            switch (screenshotFormat.Value)
+                            {
+                                case ScreenshotFormat.Bmp:
+                                    screenshotBitmap.Save(fileName, ImageFormat.Bmp);
+                                    break;
+                                case ScreenshotFormat.Png:
+                                    screenshotBitmap.Save(fileName, ImageFormat.Png);
+                                    break;
+                                case ScreenshotFormat.Jpg:
+                                    screenshotBitmap.Save(fileName, ImageFormat.Jpeg);
+                                    break;
+                                default:
+                                    throw new ArgumentOutOfRangeException(nameof(screenshotFormat));
+                            }
+                        };
+
+                    RequestScreenshot();
+                    return true;
             }
 
             return false;
         }
 
+        private string getScreenshotFileName(ScreenshotFormat screenshotFormat)
+        {
+            // TODO Change screenshots location
+            var baseDirectory = new DirectoryInfo(AppDomain.CurrentDomain.BaseDirectory);
+            var screenshotsDirectory = baseDirectory.CreateSubdirectory("Screenshots");
+
+            var screenshotExtension = screenshotFormat.ToString().ToLower();
+            var screenshots = screenshotsDirectory.GetFiles($"*.{screenshotExtension}");

```

Comment:
this will slow to a crawl. the current screenshot index should probably be stored to config.

Reply:
Or just use the current date + time. This is what other games like ffxiv do.

Reply:
I'm fine with this, as long as we choose a binary sortable format

Reply:
Date+Time, in any combination, is inherently sortable. Curious though, why you'd want this to be "binary" sortable?

Reply:
Right, sure. The ISO `yyyy-MM-dd-HH-mm-ss` standard should be the format to use - it is sortable by all file explorers. With an additional `-n` parameter at the end that increases for duplicates of screenshots taken within the same second.